cpp/include/ray/api.h

```diff
@@ -153,38 +148,29 @@ inline WaitResult Ray::Wait(const std::vector<ObjectID> &ids, int num_objects,
 
 template <typename FuncType>
 inline TaskCaller<FuncType> Ray::TaskInternal(FuncType &func) {
-  RemoteFunctionPtrHolder ptr{};
-  ptr.function_pointer = reinterpret_cast<uintptr_t>(func);
-  if (ray::api::RayConfig::GetInstance()->use_ray_remote) {
-    auto function_name = ray::internal::FunctionManager::Instance().GetFunctionName(func);
-    if (function_name.empty()) {
-      throw RayException(
-          "Function not found. Please use RAY_REMOTE to register this function.");
-    }
-    ptr.function_name = std::move(function_name);
+  RemoteFunctionHolder ptr{};
+  auto function_name = ray::internal::FunctionManager::Instance().GetFunctionName(func);
+  if (function_name.empty()) {
+    throw RayException(
+        "Function not found. Please use RAY_REMOTE to register this function.");
   }
+  ptr.function_name = std::move(function_name);
   return TaskCaller<FuncType>(ray::internal::RayRuntime().get(), ptr);
 }
 
-template <typename ActorType, typename FuncType, typename ExecFuncType>
-inline ActorCreator<ActorType> Ray::CreateActorInternal(FuncType &create_func,
-                                                        ExecFuncType &exec_func) {
-  RemoteFunctionPtrHolder ptr{};
-  if (ray::api::RayConfig::GetInstance()->use_ray_remote) {
-    auto function_name =
-        ray::internal::FunctionManager::Instance().GetFunctionName(create_func);
-    if (function_name.empty()) {
-      throw RayException(
-          "Function not found. Please use RAY_REMOTE to register this function.");
-    }
-
-    ptr.function_name = std::move(function_name);
-  } else {
-    ptr.function_pointer = reinterpret_cast<uintptr_t>(create_func);
-    ptr.exec_function_pointer = reinterpret_cast<uintptr_t>(exec_func);
+template <typename FuncType>
+inline ActorCreator<FuncType> Ray::CreateActorInternal(FuncType &create_func) {
+  RemoteFunctionHolder ptr{};
+  auto function_name =
+      ray::internal::FunctionManager::Instance().GetFunctionName(create_func);
+  if (function_name.empty()) {
+    throw RayException(
+        "Function not found. Please use RAY_REMOTE to register this function.");
   }
 
-  return ActorCreator<ActorType>(ray::internal::RayRuntime().get(), ptr);
+  ptr.function_name = std::move(function_name);
+
```

Comment:
Could you create a helper method to convert `FuncType` to `RemoteFunctionHolder` to avoid duplicate code?